include/openssl/ssl.h

```diff
@@ -1315,6 +1318,10 @@ __owur const SSL_CIPHER *SSL_get_current_cipher(const SSL *s);
 __owur int SSL_CIPHER_get_bits(const SSL_CIPHER *c, int *alg_bits);
 __owur const char *SSL_CIPHER_get_version(const SSL_CIPHER *c);
 __owur const char *SSL_CIPHER_get_name(const SSL_CIPHER *c);
+__owur int SSL_CIPHER_get_overhead(const SSL_CIPHER *c,
+				   unsigned int *int_overhead,
+				   unsigned int *blocksize,
+				   unsigned int *ext_overhead);
```

Comment:
I think it's better to make this a function of the SSL rather than the SSL_CIPHER. A cipher's overhead varies by TLS version. Block ciphers in TLS 1.1 have an explicit IV while TLS 1.0 does not. AES-GCM in TLS 1.3 will have a different construction from TLS 1.2 and lose the explicit nonce.

You might also consider reporting a "max overhead" rather than try to slice it up this way. It's less precise for the legacy CBC-mode ciphers, but a simpler interface. I dunno if we'd ever gain a variable-length AEAD in the future (would require TLS 1.3 since 1.2's AD construction doesn't allow it), but that would also generalize better if we ever did. If you prefer to, it also lets you include the record header and stuff like TLS 1.3's slightly funny encrypted content type byte which affects these computations, rather than make callers be aware of all that.